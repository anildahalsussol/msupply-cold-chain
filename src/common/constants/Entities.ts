export enum ENTITIES {
  SENSOR = 'Sensor',
  TEMPERATURE_LOG = 'TemperatureLog',
  TEMPERATURE_BREACH = 'TemperatureBreach',
  TEMPERATURE_BREACH_CONFIGURATION = 'TemperatureBreachConfiguration',
  SENSOR_LOG = 'SensorLog',
  SETTING = 'Setting',
  SYNC_LOG = 'SyncLog',
  SYNC_CONFIG = 'SyncConfig',
}
