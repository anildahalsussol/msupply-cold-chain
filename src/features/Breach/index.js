export { BreachManager } from './BreachManager';
export { BreachAction, BreachReducer, BreachSelector, BreachSaga } from './BreachSlice';
