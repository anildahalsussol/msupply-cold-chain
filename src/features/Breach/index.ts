export { BreachReducer, BreachSaga } from './BreachSlice';

export {
  CumulativeBreachAction,
  CumulativeBreachSelector,
  CumulativeBreachSaga,
  CumulativeBreachReducer,
  CumulativeBreachManager,
  CumulativeBreachLookup,
} from './CumulativeBreach';

export {
  ConsecutiveBreachAction,
  ConsecutiveBreachSelector,
  ConsecutiveBreachSaga,
  ConsecutiveBreachReducer,
  ConsecutiveBreachManager,
} from './ConsecutiveBreach';

export {
  AcknowledgeBreachAction,
  AcknowledgeBreachSelector,
  AcknowledgeBreachSaga,
  AcknowledgeBreachReducer,
  AcknowledgeBreachManager,
} from './AcknowledgeBreach';
