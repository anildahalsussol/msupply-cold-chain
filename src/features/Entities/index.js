export {
  BreachConfigurationReducer,
  BreachConfigurationAction,
  BreachConfigurationSelector,
  BreachConfigurationSaga,
  BreachConfigurationManager,
} from './BreachConfiguration';

export { SensorReducer, SensorAction, SensorSelector, SensorSaga, SensorManager } from './Sensor';

export {
  SettingReducer,
  SettingAction,
  SettingSelector,
  SettingSaga,
  SettingManager,
} from './Setting';

export { EntitiesReducer, EntitiesSaga } from './EntitiesSlice';
