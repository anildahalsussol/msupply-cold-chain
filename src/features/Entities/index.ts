export {
  BreachConfigurationReducer,
  BreachConfigurationAction,
  BreachConfigurationSelector,
  BreachConfigurationSaga,
  BreachConfigurationManager,
} from './BreachConfiguration';

export {
  SensorInitialState,
  SensorReducer,
  SensorAction,
  SensorSelector,
  SensorSaga,
  SensorManager,
} from './Sensor';

export { TemperatureLogManager } from './TemperatureLog';

export {
  SettingReducer,
  SettingAction,
  SettingSelector,
  SettingSaga,
  SettingManager,
} from './Setting';

export { EntitiesReducer, EntitiesSaga } from './EntitiesSlice';
