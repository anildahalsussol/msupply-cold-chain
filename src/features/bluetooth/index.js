export { BluetoothReducer2, BluetoothSaga } from './bluetoothSlice';
