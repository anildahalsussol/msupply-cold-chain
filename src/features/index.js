export {
  BluetoothReducer,
  BluetoothSaga,
  ScanAction,
  ScanSelector,
  ScanSaga,
  ScanReducer,
  ProgramAction,
  ProgramSelector,
  ProgramSaga,
  ProgramReducer,
  DownloadAction,
  DownloadSelector,
  DownloadSaga,
  DownloadReducer,
  DownloadManager,
  BlinkAction,
  BlinkSelector,
  BlinkSaga,
  BlinkReducer,
  BatteryObserverAction,
  BatteryObserverSelector,
  BatteryObserverSaga,
  BatteryObserverReducer,
} from './Bluetooth';

export {
  CumulativeBreachAction,
  CumulativeBreachSelector,
  CumulativeBreachSaga,
  CumulativeBreachReducer,
  CumulativeBreachManager,
  BreachReducer,
  BreachSaga,
  BreachManager,
  ConsecutiveBreachAction,
  ConsecutiveBreachSelector,
  ConsecutiveBreachSaga,
  ConsecutiveBreachReducer,
  ConsecutiveBreachManager,
} from './Breach';

export { ChartManager, ChartReducer, ChartAction, ChartSaga, ChartSelector } from './Chart';

export {
  EntitiesReducer,
  EntitiesSaga,
  SettingReducer,
  SettingAction,
  SettingSelector,
  SettingSaga,
  SettingManager,
  BreachConfigurationReducer,
  BreachConfigurationAction,
  BreachConfigurationSelector,
  BreachConfigurationSaga,
  BreachConfigurationManager,
  SensorReducer,
  SensorAction,
  SensorSelector,
  SensorSaga,
  SensorManager,
} from './Entities';

export {
  LogTableManager,
  LogTableAction,
  LogTableReducer,
  LogTableSaga,
  LogTableSelector,
} from './LogTable';

export {
  PermissionAction,
  PermissionReducer,
  PermissionSaga,
  PermissionSelector,
} from './Permission';

export { ReportManager, ReportAction, ReportReducer, ReportSaga, ReportSelector } from './Report';

export {
  SensorStatusManager,
  SensorStatusAction,
  SensorStatusReducer,
  SensorStatusSaga,
  SensorStatusSelector,
} from './SensorStatus';
