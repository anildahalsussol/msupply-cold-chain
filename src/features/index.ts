export {
  BluetoothReducer,
  BluetoothSaga,
  ScanAction,
  ScanSelector,
  ScanSaga,
  ScanReducer,
  ProgramAction,
  ProgramSelector,
  ProgramSaga,
  ProgramReducer,
  DownloadAction,
  DownloadSelector,
  DownloadSaga,
  DownloadReducer,
  DownloadManager,
  BlinkAction,
  BlinkSelector,
  BlinkSaga,
  BlinkReducer,
  BatteryObserverInitialState,
  BatteryObserverAction,
  BatteryObserverSelector,
  BatteryObserverSaga,
  BatteryObserverReducer,
} from './Bluetooth';

export {
  CumulativeBreachAction,
  CumulativeBreachSelector,
  CumulativeBreachSaga,
  CumulativeBreachReducer,
  CumulativeBreachManager,
  BreachReducer,
  BreachSaga,
  ConsecutiveBreachAction,
  ConsecutiveBreachSelector,
  ConsecutiveBreachSaga,
  ConsecutiveBreachReducer,
  ConsecutiveBreachManager,
  AcknowledgeBreachAction,
  AcknowledgeBreachSelector,
  AcknowledgeBreachSaga,
  AcknowledgeBreachReducer,
  AcknowledgeBreachManager,
} from './Breach';

export { ChartManager, ChartReducer, ChartAction, ChartSaga, ChartSelector } from './Chart';

export {
  EntitiesReducer,
  EntitiesSaga,
  SettingReducer,
  SettingAction,
  SettingSelector,
  SettingSaga,
  SettingManager,
  BreachConfigurationReducer,
  BreachConfigurationAction,
  BreachConfigurationSelector,
  BreachConfigurationSaga,
  BreachConfigurationManager,
  SensorReducer,
  SensorAction,
  SensorSelector,
  SensorSaga,
  SensorManager,
} from './Entities';

export {
  PermissionAction,
  PermissionReducer,
  PermissionSaga,
  PermissionSelector,
} from './Permission';

export { ReportManager, ReportAction, ReportReducer, ReportSaga, ReportSelector } from './Report';

export {
  SensorStatusManager,
  SensorStatusAction,
  SensorStatusReducer,
  SensorStatusSaga,
  SensorStatusSelector,
} from './SensorStatus';

export {
  SensorDetailReducer,
  SensorDetailSaga,
  DetailAction,
  DetailSelector,
  DetailReducer,
  DetailSaga,
  LogTableAction,
  LogTableManager,
  LogTableSelector,
  LogTableSaga,
  LogTableReducer,
  DetailCumulativeAction,
  DetailCumulativeSaga,
  DetailCumulativeSelector,
  DetailCumulativeReducer,
  DetailChartAction,
  DetailChartReducer,
  DetailChartSaga,
  DetailChartSelector,
} from './SensorDetail';
