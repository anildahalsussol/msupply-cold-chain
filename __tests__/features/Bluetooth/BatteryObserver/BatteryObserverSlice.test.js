import { expectSaga } from 'redux-saga-test-plan';
import { put, getContext } from 'redux-saga/effects';
import {
  BatteryObserverInitialState,
  BatteryObserverReducer,
  BatteryObserverAction,
  BatteryObserverSaga,
} from '~features/Bluetooth';

import { DEPENDENCY } from '~constants';
import { SensorAction } from '../../../../src/features/Entities';

describe('BatteryObserverSlice: BatteryObserverSaga', () => {
  it('Does not manipulate state', () => {
    const mockSensors = [{ id: '1', macAddress: 'AA:BB:CC:DD:EE:FF', batteryLevel: 90 }];
    const sensorManager = { getAll: () => mockSensors };
    const getSensorManager = () => sensorManager;
    const depsLocator = { get: getSensorManager };

    return expectSaga(BatteryObserverSaga.updateBatteryLevels)
      .provide([[getContext(DEPENDENCY.LOCATOR), depsLocator]])
      .run()
      .then(result => {
        const { effects } = result;

        expect(effects.put[0]).toEqual(put(SensorAction.update('1', 'batteryLevel', 90)));
        expect(effects.put[1]).toEqual(put(BatteryObserverAction.updateSuccess()));
        expect(effects.put.length).toEqual(2);
      });
  });
});

describe('BatteryObserverSlice: BatteryObserverReducer', () => {
  it('Case: start', () => {
    const state = BatteryObserverReducer(BatteryObserverInitialState, BatteryObserverAction.start);
    expect(state).toEqual({ isWatching: true });
  });
  it('Case: stop', () => {
    const state = BatteryObserverReducer(BatteryObserverInitialState, BatteryObserverAction.stop);
    expect(state).toEqual({ isWatching: false });
  });
});

describe('BatteryObserverSlice: BatteryObserverAction', () => {
  it('Match Snapshots', () => {
    const actionCreators = Object.values(BatteryObserverAction);

    const actions = actionCreators.map(actionCreator => actionCreator());
    expect(actions).toMatchSnapshot();
  });
});
