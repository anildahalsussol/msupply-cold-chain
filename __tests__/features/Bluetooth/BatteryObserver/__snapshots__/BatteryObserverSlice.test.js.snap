// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BatteryObserverSlice: BatteryObserverAction Match Snapshots 1`] = `
Array [
  Object {
    "payload": undefined,
    "type": "batteryObserver/start",
  },
  Object {
    "payload": undefined,
    "type": "batteryObserver/stop",
  },
  Object {
    "payload": undefined,
    "type": "batteryObserver/updateSuccess",
  },
  Object {
    "payload": undefined,
    "type": "batteryObserver/updateFail",
  },
]
`;
